devicetypes/smartthings/zigbee-accessory-dimmer.src/zigbee-accessory-dimmer.groovy

```diff
@@ -56,39 +72,57 @@ def parse(String description) {
 		}
 	} else {
 		def descMap = zigbee.parseDescriptionAsMap(description)
-		if (descMap && descMap.clusterInt == 0x0006) {
-			if (descMap.commandInt == 0x01 || descMap.commandInt == 0x00) {
+		if (descMap && descMap.clusterInt == zigbee.ONOFF_CLUSTER) {
+			if (descMap.commandInt == ONOFF_ON_COMMAND || descMap.commandInt == ONOFF_OFF_COMMAND) {
 				if (device.currentValue("level") == 0) {
 					sendEvent(name: "level", value: STEP)
 				}
-				sendEvent(name: "switch", value: device.currentValue("switch") == "on" ? "off" : "on")
+
+				if (isIkeaDimmer()) {
+					sendEvent(name: "switch", value: descMap.commandInt == ONOFF_OFF_COMMAND ? "off" : "on")
+				} else {
+					sendEvent(name: "switch", value: device.currentValue("switch") == "on" ? "off" : "on")
+				}
 			}
-		} else if (descMap && descMap.clusterInt == 0x0008) {
+		} else if (descMap && descMap.clusterInt == zigbee.LEVEL_CONTROL_CLUSTER) {
 			def currentLevel = device.currentValue("level") as Integer ?: 0
-			if (descMap.commandInt == 0x02) {
+
+			if (descMap.commandInt == LEVEL_STEP_COMMAND) {
 				def value = Math.min(currentLevel + STEP, 100)
+
 				log.debug "move to ${descMap.data}"
+
 				if (descMap.data[0] == "00") {
 					log.debug "move up"
+
 					sendEvent(name: "switch", value: "on")
 					sendEvent(name: "level", value: value)
 				} else if (descMap.data[0] == "01") {
 					log.debug "move down"
+
 					value = Math.max(currentLevel - STEP, 0)
+
 					// don't change level if switch will be turning off
 					if (value == 0) {
 						sendEvent(name: "switch", value: "off")
 					} else {
 						sendEvent(name: "level", value: value)
 					}
 				}
-			} else if (descMap.commandInt == 0x01) {
+			} else if (descMap.commandInt == LEVEL_MOVE_COMMAND) {
 				sendEvent(name: "level", value: descMap.data[0] == "00" ? 100 : STEP)
 				sendEvent(name: "switch", value: "on" )
+
 				log.debug "step to ${descMap.data}"
-			} else if (descMap.commandInt == 0x03) {
+			} else if (descMap.commandInt == LEVEL_STOP_COMMAND) {
 				log.debug "stop move"
-			}
+			} else if (descMap.commandInt == LEVEL_MOVE_LEVEL_ONOFF_COMMAND) {
```

Comment:
When does it send this command?

Reply:
This seems to be what it sends for the on and off events.